common/txmgr/types/client.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// A generic client interface for communication with the RPC node
+// Every native chain must implement independently
+type Client[
+	CHAINID ID,
+	SEQ Sequence, // nonce
+	ADDR types.Hashable,
+	BLOCK any,
+	TX any,
+	HASH types.Hashable,
```

Comment:
Can you change it to TX_HASH?
We use TX_HASH and BLOCK_HASH is many places, and it will be consistent with that naming.
Besides, HASH doesn't tell if it is TxHash or BlockHash, and the reader has to lookup to find out.

Reply:
Ohh, actually you need 2 different HASHES here.
TX_HASH types.Hashable,
BLOCK_HASH types.Hashable,

Then below in the interface, you need this:
	Transactions[TX, TX_HASH, TXRECEIPT]
	Events[EVENT, EVENTOPS]
	Blocks[BLOCK, BLOCK_HASH]